### PR TITLE
feat:  Allow temporarily disabling a key

### PR DESCRIPTION
**Pull request by MichaelUnkey**
## What does this PR do?
Eng-232

"enabled" column was added to the keys table. Web and api were updated to allow use of this boolean. 

Fixes # (issue)

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Test api to make sure all functions correctly 
Check web to make sure new feature works as intended
Check to make sure docs are updated correctly

- Test A
- Test B

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated
Screenshot 2024-01-05 at 8 15 15 AM
Screenshot 2024-01-05 at 8 15 28 AM
- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Unkey Docs if changes were necessary

### CODE REVIEW

**Review by perkinsjr — APPROVED** [review]

File: apps/api/src/pkg/keys/service.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ type VerifyKeyResult =
     }
   | {
       valid: false;
-      code: "FORBIDDEN" | "RATE_LIMITED" | "USAGE_EXCEEDED";
+      code: "FORBIDDEN" | "RATE_LIMITED" | "USAGE_EXCEEDED" | "UNAUTHORIZED";
```
[the remark this answers is not in the document]
Comment (chronark) [reply apps/api/src/pkg/keys/service.ts:24]: @MichaelUnkey let's go with `DISBABLED` :)

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/api/src/pkg/keys/service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichaelUnkey) [reply apps/api/src/pkg/keys/service.ts:24]: Added as requested

File: apps/api/src/routes/v1_keys_updateKey.ts [hunk 2]
```diff
@@ -210,6 +215,7 @@ export const registerV1KeysUpdate = (app: App) =>
           refillInterval: req.refill === null ? null : req.refill?.interval,
           refillAmount: req.refill === null ? null : req.refill?.amount,
           lastRefillAt: req.refill == null || req.refill?.amount == null ? null : new Date(),
+          enabled: req.enabled === null ? undefined : req.enabled,
```
[the remark this answers is not in the document]
Comment (MichaelUnkey) [reply apps/api/src/routes/v1_keys_updateKey.ts:218]: Added tests and made requested changes

File: apps/web/components/dashboard/api-key-table/index.tsx [hunk 3]
```diff
@@ -122,6 +131,20 @@ export const ApiKeyTable: React.FC<Props> = ({ data }) => {
       ),
       cell: ({ row }) => row.original.createdAt.toUTCString(),
     },
+    {
+      accessorKey: "enabled",
+      header: "Enabled",
+      cell: ({ row }) =>
+        row.original.enabled ? (
+          <span>
+            <CheckCircle2 />
```
[the remark this answers is not in the document]
Comment (MichaelUnkey) [reply apps/web/components/dashboard/api-key-table/index.tsx]: changed

File: apps/web/components/dashboard/api-key-table/index.tsx [hunk 4]
```diff
@@ -122,6 +131,20 @@ export const ApiKeyTable: React.FC<Props> = ({ data }) => {
...
+          </span>
+        ) : (
+          <span>
+            <XCircle />
```
[the remark this answers is not in the document]
Comment (MichaelUnkey) [reply apps/web/components/dashboard/api-key-table/index.tsx]: Changed

File: internal/db/src/schema/keys.ts [hunk 5]
```diff
@@ -55,6 +56,11 @@ export const keys = mysqlTable(
     refillInterval: mysqlEnum("refill_interval", ["daily", "monthly"]),
     refillAmount: int("refill_amount"),
     lastRefillAt: datetime("last_refill_at", { fsp: 3 }),
+    /**
+     * sets if key is enabled or disabled
+     */
+    enabled: boolean("enabled").default(true).notNull(),
```
[the remark this answers is not in the document]
Comment (chronark) [reply internal/db/src/schema/keys.ts:62]: looks good to go

**Review by chronark — APPROVED** [review]

**Pull request merged**